src/tools/compiletest/src/runtest.rs

```diff
@@ -468,7 +468,13 @@ impl<'test> TestCx<'test> {
 
         if let Some(revision) = self.revision {
             let normalized_revision = normalize_revision(revision);
-            cmd.args(&["--cfg", &normalized_revision]);
+            let cfg_arg = ["--cfg", &normalized_revision];
+            if self.props.compile_flags.windows(2).any(|args| args == cfg_arg) {
+                panic!(
+                    "warning: redundant cfg argument `{normalized_revision}` is already created by the revision"
+                );
+            }
```

Comment:
cfgs can also be set from a single argument, i.e. `--cfg=foo`

Reply:
Ah right, good point.